src/processing/dom/index.js

```diff
@@ -57,55 +65,61 @@ export default class DomProcessor {
     _createProcessorPatterns (adapter) {
         var selectors = {
             HAS_HREF_ATTR: el => {
-                var tagName = adapter.getTagName(el).toLowerCase();
+                var tagName = adapter.getTagName(el);
 
                 return URL_ATTR_TAGS.href.indexOf(tagName) !== -1;
             },
 
             HAS_SRC_ATTR: el => {
-                var tagName = adapter.getTagName(el).toLowerCase();
+                var tagName = adapter.getTagName(el);
 
                 return URL_ATTR_TAGS.src.indexOf(tagName) !== -1;
             },
 
             HAS_ACTION_ATTR: el => {
-                var tagName = adapter.getTagName(el).toLowerCase();
+                var tagName = adapter.getTagName(el);
 
                 return URL_ATTR_TAGS.action.indexOf(tagName) !== -1;
             },
 
             HAS_MANIFEST_ATTR: el => {
-                var tagName = adapter.getTagName(el).toLowerCase();
+                var tagName = adapter.getTagName(el);
 
                 return URL_ATTR_TAGS.manifest.indexOf(tagName) !== -1;
             },
 
             HAS_DATA_ATTR: el => {
-                var tagName = adapter.getTagName(el).toLowerCase();
+                var tagName = adapter.getTagName(el);
 
                 return URL_ATTR_TAGS.data.indexOf(tagName) !== -1;
             },
 
             HTTP_EQUIV_META: el => {
-                var tagName = adapter.getTagName(el).toLowerCase();
+                var tagName = adapter.getTagName(el);
 
                 return tagName === 'meta' && adapter.hasAttr(el, 'http-equiv');
             },
 
             ALL: () => true,
 
-            IS_SCRIPT: el => adapter.getTagName(el).toLowerCase() === 'script',
+            IS_SCRIPT: el => adapter.getTagName(el) === 'script',
 
-            IS_LINK: el => adapter.getTagName(el).toLowerCase() === 'link',
+            IS_LINK: el => adapter.getTagName(el) === 'link',
 
-            IS_INPUT: el => adapter.getTagName(el).toLowerCase() === 'input',
+            IS_INPUT: el => adapter.getTagName(el) === 'input',
 
-            IS_STYLE: el => adapter.getTagName(el).toLowerCase() === 'style',
+            IS_STYLE: el => adapter.getTagName(el) === 'style',
 
             HAS_EVENT_HANDLER: el => adapter.hasEventHandler(el),
 
-            IS_SANDBOXED_IFRAME: el => adapter.getTagName(el).toLowerCase() === 'iframe' &&
-                                       adapter.hasAttr(el, 'sandbox')
+            IS_SANDBOXED_IFRAME: el => adapter.getTagName(el) === 'iframe' &&
+                                       adapter.hasAttr(el, 'sandbox'),
+
+            IS_SVG_ELEMENT_WITH_XLINK_HREF_ATTR: el => {
+                return adapter.hasAttr(el, 'xlink:href') &&
+                       SVG_XLINK_HREF_TAGS.indexOf(adapter.getTagName(el)) !== -1 &&
+                       adapter.isSVGElement(el);
```

Comment:
I'd rather move `adapter.isSVGElement(el)` condition to the first place, since it's the most assertive and low-cost check.